pkg/storage/replica.go

```diff
@@ -4830,7 +4830,8 @@ func (r *Replica) evaluateTxnWriteBatch(
 	ms := enginepb.MVCCStats{}
 	// If not transactional or there are indications that the batch's txn will
 	// require restart or retry, execute as normal.
-	if !r.store.TestingKnobs().DisableOnePhaseCommits && isOnePhaseCommit(ba) {
+	allowed1PC, required1PC := onePhaseCommitStatus(ba)
```

Comment:
It's odd to have to think about the combination `(false, true)`. Perhaps pass `DisableOnePhaseCommits` into `onePhaseCommitStatus` and then return a single bool which indicates whether to use one phase commit? And consider renaming: `DisableOptionalOnePhaseCommits`.